platform/view/services/comm/pki.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package comm
+
+import (
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/pem"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/hash"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type PKISupport struct{}
+
+func (p PKISupport) ExtractPublicKey(id view.Identity) any {
+	certRaw, _ := pem.Decode(id)
+	switch {
```

Comment:
Maybe replacing the switch statement with a nil check at the beginning could be more readable:
```
certRaw, _ := pem.Decode(id)
if certRaw == nil {
 return nil
}
cert, err := x509.ParseCertificate(certRaw.Bytes)
...
```

Reply:
👍